### ISSUE TITLE
No way to exclude magical items from search.

### ISSUE THREAD

**Issue by Nikkotg** [MrXofar/compendium-navigator#3]
While the navigator is great, having the option to exclude magical items would be very helpful. Maybe a checkbox for "magical" or not. Trying to find mundane weapons or equipment is a challenge when it shows everything available.

**Comment by MrXofar** [comment]
Which system are you using?

**Comment by Nikkotg** [comment]
Dnd5e. I've since taken a deeper look at item "properties" and 
"mgc": true,
is applied to magic items and referenced in in the navigator filters. It's just that it only filters for those with the tag and either doesn't allow exclusion or I just haven't found a way to so.

**Comment by MrXofar** [comment]
Was just looking at that field. Need a third state of the checkboxes to exclude. 
![image](https://user-images.githubusercontent.com/103948142/234997763-ff0b7f9c-f186-48c2-9071-db897ba59461.png)

**Comment by MrXofar** [comment]
Something like this?
Unchecked: Filter all
Checked: Only include
Grey Check: Exclude

Hmmm

**Comment by Nikkotg** [comment]
You'd know better than me lol. My highschool level javascript is all out the window. As for interface that sounds good to me. I am curious what it takes to translate the imported data fields and relabel them to a more readable format. ".mgc" among the other default names can be easy to overlook and hard to parse. But that's beyond the scope of this issue. Thanks for the prompt replies

**Comment by MrXofar** [comment]
Yeah, translating them all, and then doing that for all supported systems would be a monster undertaking. I found this link that gives me an idea on how to accomplish this. Thanks for the input.
https://carsonf92.medium.com/introducing-the-three-state-checkbox-1b6f00b6ec89

**Comment by Nikkotg** [comment]
If it's a fairly copy-paste formula to rename parameter fields, minus changing the titles to fit, I wouldn't mind offering to help with some grunt work. Although my credentials are just value tweaking a fishing minigame module for my own use

**Comment by MrXofar** [comment]
I could possibly see using a json file with key value pairs naming the key path and friendly name. So, something like this might work, and I could just reference that when rendering the labels.

      "item.weapon": {
        "system.weaponType": "Weapon Type",
        "system.properties.amm": "Ammunition",
        "system.properties.mgc": "Magical"      
      }

Might not even need to do every single property, just the ones that are not reader friendly, or don't really communicate what they are - like "system.damage.parts.0.1". I probably wouldn't need one for "system.properties.weaponType" because it is self explanatory.

**Comment by Nikkotg** [comment]
That's pretty clever. Would love to see this module progress to be a bit more friendly to players and people who don't have much time or knowhow to get familiar. Probably the best compendium filter I've seen so far

**Comment by MrXofar** [comment]
Thanks, I really appreciate the positive feedback.  You should check out my other module called [Roll New Character Stats](https://foundryvtt.com/packages/roll-new-character-stats)

I will keep this issue open so I don't forget about it. Will work on it when time permits.
